src/osx/cocoa/dataview.mm

```diff
@@ -2320,64 +2324,71 @@ void UpdateWithRow(int row)
     // visible miscalculations, we also include all currently visible items
     // no matter what.  Finally, the value of N is determined dynamically by
     // measuring how much time we spent on the determining item widths so far.
-
+    if (calculateAllRows)
```

Comment:
Is the condition here correct? It looks like we're using the optimized version for "all rows" case, when I'd expect it to iterate over all rows. Or maybe the variable is named incorrectly instead?

Reply:
Trying to explain on behalf of @dkulp, so please correct me if I got something wrong…

These changes are trying to fix previously merged changes to this code (that IMHO in retrospect shouldn't have been merged). Part of the problem with them was that they changed `FitColumnWidthToContent()` to do much more than the name says —  on its face, it still does column width fitting, but *also* calculates (max) row height, and column expander width. `FitColumnWidthToContent` is no longer a fitting name. `MaxWidthCalculator` is now misnamed too. And the function is, because of that change of purpose, called much more often than it used to.

This is a problem, because it kills performance. So @dkulp is trying to limit pointless calculations. One of the optimizations is to avoid iterating over all the rows if we know it won't provide any new information.

So yes, it is a badly named variable. It should be e.g. `shouldCalculateRows`.